libs/utils/trace.py

```diff
@@ -195,6 +203,41 @@ def __computeTimeSpan(self):
             logging.info('Overutilized time: %.6f [s] (%.3f%% of trace time)',
                     self.overutilized_time, self.overutilized_prc)
 
+    def _loadFunctionsStats(self, path='trace.stats'):
+        if os.path.isdir(path):
+            path = os.path.join(path, 'trace.stats')
+        if path.endswith('dat'):
+            pre, ext = os.path.splitext(path)
+            stats_file = pre + '.stats'
+        if not os.path.isfile(path):
+            return False
+
+        # Opening functions profiling JSON data file
+        logging.debug('Loading functions profiling data from [%s]...', path)
+        with open(os.path.join(path), 'r') as fh:
+            trace_stats = json.load(fh)
+
+        # Build DataFrame of function stats
+        frames = {}
+        for cpu, data in trace_stats.iteritems():
+            frames[int(cpu)] = pd.DataFrame.from_dict(data, orient='index')
+
+        # Build and keep track of the DataFrame
+        self._functions_stats_df = pd.concat(frames.values(), keys=frames.keys())
+
+        return len(self._functions_stats_df) > 0
+
+    def functions_stats_df(self, functions=None):
+        if not hasattr(self, '_functions_stats_df'):
+            return None
+        df = self._functions_stats_df
+        # Filter data for required functions names
+        if functions:
```

Comment:
I think `functions` must be a required argument. I don't see a point in calling this with `None`.